LEGO1/omni/src/stream/mxramstreamprovider.cpp

```diff
@@ -105,14 +105,16 @@ MxResult MxRAMStreamProvider::SetResourceToGet(MxStreamController* p_resource)
 // FUNCTION: BETA10 0x1016492f
 MxU32 ReadData(MxU8* p_buffer, MxU32 p_size)
 {
+	MxU32 fourcc;
 	MxU32 id;
 	MxU8* data = p_buffer;
 	MxU8* data2;
 
 #define IntoType(p) ((MxU32*) (p))
 
 	while (data < p_buffer + p_size) {
-		if (*IntoType(data) == FOURCC('M', 'x', 'O', 'b')) {
+		memcpy(&fourcc, IntoType(data), sizeof(MxU32));
+		if (fourcc == FOURCC('M', 'x', 'O', 'b')) {
```

Comment:
This one needs extra attention, without it my hunt for UB stops here.
With it ASAN yells at me: unknown-crash

Reply:
And without ASAN i get this:
```
terminate called after throwing an instance of 'std::bad_array_new_length'
  what():  std::bad_array_new_length
```
The only remaining option is using a debugger and I don't find that fun so it's up to someone else :P

Reply:
Managed to solve both issues